File: 2022/day1.1.js
function solvePuzzle(input) {
	const rows = input.split('\n')

	let elfKcal = 0
	let greatestKcalTotal = 0

	for (const row of rows) {
		const kcal = Number(row)

		if (kcal) {
			elfKcal += kcal
		} else {
			greatestKcalTotal = Math.max(elfKcal, greatestKcalTotal)
			elfKcal = 0
		}
	}

	return greatestKcalTotal
}

const input = document.querySelector('pre').innerText
const result = solvePuzzle(input)

console.log(result)

File: 2022/day1.1.py
with open("2022/input.txt", "r") as file:
    input = file.read()


def solve_puzzle(input):
    highest_kcal_total = 0

    for elf_foods in input.split("\n\n"):
        kcals = map(int, elf_foods.split("\n"))
        elf_total = sum(kcals)
        highest_kcal_total = max(highest_kcal_total, elf_total)

    return highest_kcal_total


result = solve_puzzle(input)

print(result)
